File: frontend/src/types/grupo.ts
//src/types/grupo.ts
export type RolGrupo = "lider" | "miembro";

export interface Grupo {
  id_grupo: number;
  nombre_grupo: string;
  descripcion: string | null;
  codigo_acceso: string;
  fecha_creacion: string;
}

export interface GrupoConRol extends Grupo {
  rol: RolGrupo;
}

export interface Integrante {
  id_integrante: number;
  id_usuario: number;
  correo_electronico: string;
  rol: RolGrupo;
  fecha_ingreso: string;
}

File: backend/app/schemas/grupo.py
from datetime import date
from typing import Optional
from pydantic import BaseModel, ConfigDict, Field


class GrupoCreate(BaseModel):
    nombre_grupo: str = Field(..., max_length=50)
    descripcion: Optional[str] = None


class GrupoResponse(BaseModel):
    id_grupo: int
    nombre_grupo: str
    descripcion: Optional[str] = None
    codigo_acceso: str
    fecha_creacion: date

    model_config = ConfigDict(from_attributes=True)


class GrupoConRolResponse(GrupoResponse):
    rol: str


class GrupoJoinRequest(BaseModel):
    codigo_acceso: str = Field(..., max_length=20)


class IntegranteResponse(BaseModel):
    id_integrante: int
    id_usuario: int
    correo_electronico: str
    rol: str
    fecha_ingreso: date

    model_config = ConfigDict(from_attributes=True)